File: backend/auth-service/src/main/java/com/apexbank/auth/service/AuthService.java
package com.apexbank.auth.service;

import com.apexbank.auth.dto.request.*;
import com.apexbank.auth.dto.response.CurrentUserResponse;
import com.apexbank.auth.dto.response.LoginResponse;
import com.apexbank.auth.dto.response.RefreshTokenResponse;

public interface AuthService {

    LoginResponse register(RegisterRequest request);

    LoginResponse login(LoginRequest request);

    CurrentUserResponse getCurrentUser();

    RefreshTokenResponse refreshToken(RefreshTokenRequest request);

    void logout(
            String accessToken,
            String refreshToken
    );
    void forgotPassword(ForgotPasswordRequest request);

    void resetPassword(ResetPasswordRequest request);

    boolean validateResetToken(String token);

}

File: apps/customer-portal/src/features/auth/services/auth.service.ts
import api from "@/services/api/axios";

import type {
  ForgotPasswordRequest,
  LoginRequest,
  RegisterRequest,
  ResetPasswordRequest,
  VerifyOtpRequest,
} from "@/types/auth/auth.request";

import type {
  LoginResponse,
} from "@/types/auth/auth.response";

import type {
  ApiResponse,
} from "@/types/common/api.response";

export const authService = {
  login(request: LoginRequest) {
    return api.post<ApiResponse<LoginResponse>>(
      "/auth/login",
      request,
    );
  },

  register(request: RegisterRequest) {
    return api.post("/auth/register", request);
  },

  forgotPassword(request: ForgotPasswordRequest) {
    return api.post("/auth/forgot-password", request);
  },

  verifyOtp(request: VerifyOtpRequest) {
    return api.post("/auth/verify-otp", request);
  },

  resetPassword(request: ResetPasswordRequest) {
    return api.post("/auth/reset-password", request);
  },
};